src/pages/_app.tsx

```diff
@@ -1,11 +1,27 @@
-import { AppProps } from "next/app";
-import { ApolloClient, InMemoryCache } from "@apollo/client";
+import { ApolloClient, concat, HttpLink, InMemoryCache } from "@apollo/client";
+import { setContext } from "@apollo/client/link/context";
 import { ApolloProvider } from "@apollo/client/react";
-import { AuthProvider } from "../../utils/firebase/auth";
+import { AppProps } from "next/app";
 import "tailwindcss/tailwind.css";
+import { AuthProvider } from "../../utils/firebase/auth";
+import firebase from "../../utils/firebase/firebase";
+
+const httpLink = new HttpLink({ uri: process.env.NEXT_PUBLIC_API_URL });
+
+const authLink = setContext(async (_, { headers }) => {
+  const token = await firebase
```

Comment:
AuthProvider has an "auth" object that we can also use. When would we be using that vs. firebase.auth().currentUser?

Reply:
We'd use `auth` in the vast majority of cases, we only use `firebase.auth().currentUser` here because the Apollo client is initialized outside of all the components.